training/controllers/user.py

```diff
@@ -0,0 +1,108 @@
+from flask import Blueprint, request, redirect, render_template, url_for, g, current_app, flash, make_response, \
+    jsonify, abort
+from flask import send_from_directory
+from werkzeug.utils import secure_filename
+import os
+
+from training.extensions import db
+from training.controllers.function_decorators import login_required
+from training.controllers.forms.upload_image_avatar_form import UploadImageForm
+
+
+ALLOWED_EXTENSIONS = {'txt', 'pdf', 'png', 'jpg', 'jpeg', 'gif'}
+
+bp = Blueprint('avatar', __name__, url_prefix='/user')
+
+
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+def save_path_to_database_user(path_to_image):
+    if len(path_to_image) > 50:
+        abort(400)
+    else:
+        if g.user.path_to_image is None:
+            g.user.path_to_image = path_to_image
+            db.session.commit()
+        else:
+            os.remove(f"uploads/{g.user.username}/{g.user.path_to_image}")
+            g.user.path_to_image = path_to_image
+            db.session.commit()
+
+
+@bp.errorhandler(400)
+def not_found(error):
+    return make_response(jsonify({'error': 'Path too long, bad request.',
+                                  "param": error}), 400)
+
+
+def rename(filename):
+    raw_email = str(g.user.email)
+    a = raw_email + "_" + filename
```

Comment:
Are you using the user's email to name the uploaded file?

Reply:
No, it is an unused function, it must be removed.
The name of the uploaded file will be the same as the user's original file, but before it is saved, it is passed to the secure_filename function.
secure_filename does the following:
Pass it a filename and it will return a secure version of it. [...] 
Example:
    >> secure_filename("My cool movie.mov")
    'My_cool_movie.mov'
    >>> secure_filename("../../../etc/passwd")
    'etc_passwd'
    >>> secure_filename(u'i contain cool \xfcml\xe4uts.txt')
    'i_contain_cool_umlauts.txt'